plotly/figure_factory/_dendrogram.py

```diff
@@ -140,7 +140,10 @@ def __init__(self, X, orientation='bottom', labels=None, colorscale=None,
         for i in range(len(yvals_flat)):
             if yvals_flat[i] == 0.0 and xvals_flat[i] not in self.zero_vals:
                 self.zero_vals.append(xvals_flat[i])
-
```

Comment:
Let's add a nice comment here explaining why this check is here. (e.g. the situation with multiple identical values)